source/protocol/lsp-generic_sets.ads

```diff
@@ -0,0 +1,40 @@
+------------------------------------------------------------------------------
+--                         Language Server Protocol                         --
+--                                                                          --
+--                     Copyright (C) 2018-2019, AdaCore                     --
+--                                                                          --
+-- This is free software;  you can redistribute it  and/or modify it  under --
+-- terms of the  GNU General Public License as published  by the Free Soft- --
+-- ware  Foundation;  either version 3,  or (at your option) any later ver- --
+-- sion.  This software is distributed in the hope  that it will be useful, --
+-- but WITHOUT ANY WARRANTY;  without even the implied warranty of MERCHAN- --
+-- TABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public --
+-- License for  more details.  You should have  received  a copy of the GNU --
+-- General  Public  License  distributed  with  this  software;   see  file --
+-- COPYING3.  If not, go to http://www.gnu.org/licenses for a complete copy --
+-- of the license.                                                          --
+------------------------------------------------------------------------------
+
+--  This package provides generic set to implement Language Server Protocol.
+
+with Ada.Streams;
+
+generic
+   type Element is (<>);
+
+package LSP.Generic_Sets is
```

Comment:
`Generic_Sets` does not seem the right name for this - is this really a set? Could we call it `Generic_Boolean_Arrays` and change the type from `Set` to `Boolean_Array` ?

Reply:
It's a set of enumerated values. It's represented as array of Boolean. So you can check if a value in a set with `if Set (Value) then`. I guess, this is well-known way to have effective set of small number of values in Ada.

Reply:
What I'm trying to do is avoid the confusion with `Set` as in the container (`Hashed_Set` and so forth). How about `Generic_Bitfield` or (by favourite) `Generic_Flags` for the package name? And call the function `Is_Set`, so clients can call `if Is_Set (Value) then`.

Reply:
But this Set **is** a container. There isn't any semantic difference between `new Generic_Sets (My_Enum)` and `new Ada.Container.Hashed_Sets (My_Enum)`. I just provide more effective implementation.

Ok. I'll hide the array in the private part and put standard `Contains`, `Include`, `Delete` subprograms into visible part, if you want.

Reply:
no - if your implementation is the same semantic as `Ada.Containers.Hashed_Sets`, then we should use an `Ada.Containers.Hashed_Sets` - no reason to add code, it's just technical debt for no reason, the speed optimization is not needed here.